src/rpc/core_rpc_server.cpp

```diff
@@ -1560,7 +1560,8 @@ namespace cryptonote
 
     for (auto & entry : white_list)
     {
-      if (!req.include_blocked && m_p2p.is_host_blocked(entry.adr, NULL))
+      if ((!req.include_blocked || !req.include_banned) 
```

Comment:
The logic here and below differ (pretty sure this one is wrong, and you want `!(included_banned || include_blocked)`; it might help to just bundle the logic for `include_banned` into a local constant, given that it's multiple-sourced.

Reply:
You are correct. The logic is wrong. Actually, I mistyped it. I had `((!req.include_blocked && !req.include_banned)` which is same as what you are suggesting. But I mis-typed it or pushed the wrong implementation. 

Anyway, thanks a lot for catching this. Will push a newer one with corrected code.